src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -376,6 +376,11 @@ def _validate_vm_create_storage_profile(namespace, for_scale_set=False):
         namespace.attach_os_disk = _get_resource_id(
             namespace.attach_os_disk, namespace.resource_group_name, 'disks', 'Microsoft.Compute')
 
+    if getattr(namespace, 'attach_data_disks', None):
```

Comment:
Actually with hybrid code for both python 2 and 3, `hasattr` has [drawbacks](https://hynek.me/articles/hasattr/). For this case, it doesn't apply, but for consistency, i just use `getattr` across